### PR TITLE
Construct UnionSDF from RobotModel 

### PR DESCRIPTION
**Pull request by HiroIshida**
This PR implements a new constructor for `UnionSDF` that takes `RobotModel` as a parameter. This constructor enables directory convert `RobotModel` to `UnionSDF`. This feature is depends on my previous PRs: https://github.com/iory/scikit-robot/pull/198, https://github.com/iory/scikit-robot/pull/201

Existing robot models such as fetch and pr2 doesn't have sphere link. But for testing, I need some model that have all primitive geometric shape. So, I added a new urdf model named r2d2 which consists of only primitive geometric shapes, which is used in the unittest. 

A demo for making UnionSDF from RobotModel. In the picture, points are sampled from the implicit surface of the UnionSDF.  
```python
import skrobot
from skrobot.model import Sphere
from skrobot.sdf import UnionSDF

fetch = skrobot.models.Fetch()
fetch.reset_manip_pose()
union = UnionSDF.from_robot_model(fetch)
pts, _ = union.surface_points(n_sample=100)

viewer = skrobot.viewers.TrimeshSceneViewer(resolution=(641, 480))
viewer.add(fetch)
viewer.show()
for pt in pts:
    sp = Sphere(radius=0.01, pos=pt, color=[255, 0, 0, 200])
    viewer.add(sp)
```

hoge

## TODO 
~~The third TODO will be done after https://github.com/iory/scikit-robot/pull/201 is merged~~
- [x] implementation
- [x] write a test
- [x] integrate CylinderSDF

### CODE REVIEW

**Comment by HiroIshida** [comment]
nice that the PR is quickly merged! 
https://github.com/mikedh/trimesh/pull/1106
I will close https://github.com/iory/scikit-robot/pull/198 and fix this PR based on your PR

**Comment by HiroIshida** [comment]
@iory I fixed the PR reflecting the update of trimesh. In `urdf.py` the information of origin (translation and rot) is removed and just made a new mesh. Thus, I added the origin information to the metadata of the each mesh. I still feel this method is kind of ad-hoc, but reasonable workaround. I want to here your opinion.

File: skrobot/sdf/signed_distance_function.py [hunk 1]
```diff
@@ -8,11 +8,66 @@
...
+    translation = origin[:3, 3]
+    sdf.coords = make_cascoords(pos=translation, rot=rotation_matrix)
+    return sdf
+
+
+def link2sdf(link, urdf_path, dim_grid=30):
+    """convert Link to corresponding sdf
+
+    Parameters
+    -------
```
Comment (iory) [inline skrobot/sdf/signed_distance_function.py:50]: Please add `-`.

File: skrobot/sdf/signed_distance_function.py [hunk 2]
```diff
@@ -8,11 +8,66 @@
...
+    link : skrobot.model.Link
+        link object
+    urdf_path : str
+        urdf path of the robot model that the link belongs to
+    dim_grid : int
+        dimension of the GridSDF
```
Comment (iory) [inline skrobot/sdf/signed_distance_function.py:56]: Is it only for `GridSDF`?

File: skrobot/sdf/signed_distance_function.py [hunk 3]
```diff
@@ -8,11 +8,66 @@
...
+
+    origin = mesh.metadata["origin"]
+    rotation_matrix = origin[:3, :3]
+    translation = origin[:3, 3]
+    sdf.coords = make_cascoords(pos=translation, rot=rotation_matrix)
+    return sdf
+
+
+def link2sdf(link, urdf_path, dim_grid=30):
+    """convert Link to corresponding sdf
```
Comment (iory) [inline skrobot/sdf/signed_distance_function.py:47]: Please capitalize.

File: skrobot/sdf/signed_distance_function.py [hunk 4]
```diff
@@ -174,6 +229,28 @@ def _surface_points(self, n_sample=1000):
...
+
+        Returns
+        -------
+        union_sdf : skrobot.sdf.UnionSDF
+            union sdf of robot_model
+        """
+        sdf_list = []
+        for link in robot_model.link_list:
+            if link.collision_mesh is not None:
+                sdf = link2sdf(link, robot_model.urdf_path, dim_grid=50)
```
Comment (iory) [inline skrobot/sdf/signed_distance_function.py:250]: `dim_grid=dim_grid`

File: tests/skrobot_tests/test_sdf.py [hunk 5]
```diff
@@ -211,3 +211,31 @@ def test_unionsdf_surface_points(self):
 
         cond_and = (sum(on_surface1) > 0) and (sum(on_surface2) > 0)
         self.assertTrue(cond_and)  # each surface has at least a single points
+
+    def test_union_sdf_from_robot_model(self):
+        r2d2 = skrobot.models.urdf.RobotModelFromURDF(
```
Comment (iory) [inline tests/skrobot_tests/test_sdf.py:216]: Why did you use new model `r2d2`?

**Review by iory — CHANGES_REQUESTED** [review]

**Comment by HiroIshida** [comment]
Fixed minor mistake in https://github.com/iory/scikit-robot/pull/199

dim_grid is only the parameter for GridSDF because other primitive's sdf can be computed analytically

the reason for adding r2d2 is that they have all primitives (fetch and PR2 are not). So testing that all primitive is properly converted to sdf, I need add new robot.

**Comment by iory** [comment]
Where did you take the original r2d2 urdf?

**Comment by HiroIshida** [comment]
Took from here: 
http://wiki.ros.org/urdf/Tutorials/Building%20a%20Visual%20Robot%20Model%20with%20URDF%20from%20Scratch
but I removed some mesh related elements stuff. 

But, I found that

File: skrobot/data/r2d2_description/r2d2.urdf [hunk 6]
```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0"?>
+<robot name="materials">
```
Comment (iory) [inline skrobot/data/r2d2_description/r2d2.urdf:2]: Is name correct?

**Review by iory — CHANGES_REQUESTED** [review]

**Comment by HiroIshida** [comment]
named is fixed to r2d2 in https://github.com/iory/scikit-robot/pull/199/commits/282cd9551589c1eb837cdcb93abe42b427f7499b

**Comment by iory** [comment]
Hmm, the model is far from `r2d2`.
Does the main role of `r2d2` is for testing `Sphere`?

**Comment by HiroIshida** [comment]
Its only for testing Sphere and Cylinder. It is also possible to make a model after loading the URDF, like, by associng and adding these primitives. However by this way, we cannot test the case when the mesh is transformed in the creation of urdf.

**Comment by iory** [comment]
`Cylinder` and `Box` are contained by `PR2`. 
If you want to add `r2d2`, please make it more sophisticated. It's not `r2d2`.

**Comment by HiroIshida** [comment]
I decided put the urdf file into the test directory as it will be used only in test and never be used by the user.

File: tests/skrobot_tests/primitives.urdf [hunk 7]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0"?>
```
Comment (iory) [inline tests/skrobot_tests/primitives.urdf:1]: Please make skrobot_tests/data directory and put this in it.

**Review by iory — CHANGES_REQUESTED** [review]

**Comment by HiroIshida** [comment]
Completed. (Sorry for directory writing in this comment box, somehow I cannot reply to the review...)

File: tests/data/primitives.urdf [hunk 8]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0"?>
```
Comment (iory) [inline tests/data/primitives.urdf:1]: `tests/skrobot_tests/data/primitives.urdf`.

**Review by iory — CHANGES_REQUESTED** [review]

**Review by iory — APPROVED** [review]

**Pull request merged**